dev-libs/smooth/smooth-0.8.74.0~pre4.recipe

```diff
@@ -0,0 +1,95 @@
+SUMMARY="A cross-platform C++ class library"
+DESCRIPTION="smooth is an object oriented C++ class library for Windows, \
+macOS, Linux and most Unix-like operating systems. It provides basic \
+functionality and platform support for applications and libraries.
+
+Features provided by smooth include:
+
+- user interface API with various widgets
+- simple to use multithreading API
+- file and network IO interface
+- completely transparent Unicode and software internationalization support
+- a libxml2 based XML parser"
+HOMEPAGE="http://www.smooth-project.org/"
+COPYRIGHT="1998-2018 Robert Kausch"
+LICENSE="Artistic v2.0"
+REVISION="1"
+SOURCE_URI="https://downloads.sourceforge.net/smooth/smooth-${portVersion/\~/-}.tar.gz"
+CHECKSUM_SHA256="815f0329b5546c78feeb9ad1813855dfcb7bc062e28de44d0b4aadd7c6625d80"
+SOURCE_DIR="smooth-${portVersion/\~/-}"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+fi
+
+libVersion="${portVersion%~*}"
+libVersionCompat="$libVersion compat >= ${libVersion%.*}"
+
+PROVIDES="
+	smooth$secondaryArchSuffix = $portVersion
+	lib:libsmooth$secondaryArchSuffix = $libVersionCompat
```

Comment:
I'm not getting any `libsmooth.so` symlink nor a `libsmooth.so.0` file. Well, I'm getting them with different names:
```
lib
  libsmooth-0.8.74.so -> libsmooth-0.8.74.so.0
  libsmooth-0.8.74.so.0
```
Traditionally we would get:
```
lib
  libsmooth.so.0 -> libsmooth.so.0.8.74
  libsmooth.so.0.8.74
```
I don't know what's the best way to handle your layout. Maybe:
```diff
-libVersion="${portVersion%~*}"
+libRealVersion="${portVersion%~*}"
-libVersionCompat="$libVersion compat >= ${libVersion%.*}"
+libVersionCompat="${libRealVersion%%.*}"
 
 PROVIDES="
 	smooth$secondaryArchSuffix = $portVersion
-	lib:libsmooth$secondaryArchSuffix = $libVersionCompat
- 	lib:libsmooth_${libVersion%.*}$secondaryArchSuffix = $libVersionCompat
+ 	lib:libsmooth_${libRealVersion%.*}$secondaryArchSuffix = $libVersionCompat
```
I think it would work, but I'm not satisfied with this.

I suspect that any package that would require the lib from the smooth package would have to be rebuilt everytime the version of smooth is bumped.

Wouldn't it be better to have a `libsmoth.so.0` symlink point to `libsmooth.so.0.8.74` ?

BTW, the devel package does not provide and symlinks for the lib. We would normally do this with:
```
	prepareInstalledDevelLib libsmooth
	fixPkgconfig
```
but here it would have to be:
```
	prepareInstalledDevelLib libsmooth-${libRealVersion%.*}
```

This is not an easy question. Let's see what other contributors think about this.

Reply:
The smooth API and ABI still change quite often, that's why I would like to keep it as `libsmooth-0.8.74.so`. A program linked against smooth 0.8.74.0 will be compatible with 0.8.74.3, but might not be compatible with smooth 0.8.75.0. I plan to keep the ABI stable when going to a 0.9 release.

I'll add the prepareInstalledDevelLib call though.